src/emulator/auth/operations.ts

```diff
@@ -944,12 +941,23 @@ export function setAccountInfoImpl(
 
     if (reqBody.email) {
       assert(isValidEmailAddress(reqBody.email), "INVALID_EMAIL");
+      if (!emulatorUrl) {
+        throw new Error("Internal assertion error: Emulator URL invalid");
+      }
+
       const newEmail = canonicalizeEmailAddress(reqBody.email);
       if (newEmail !== user.email) {
         assert(!state.getUserByEmail(newEmail), "EMAIL_EXISTS");
```

Comment:
BLOCKING: Is it an error to set your email to someone else's `initialEmail`? Please double check production behavior and add a test case. If you can, then what happens if that person reverts their email using an oobCode? Would that violate the uniqueness constraint of `user.email`?

Reply:
Good call. You can set your email to someone else's `initialEmail`, but it will fail with `EMAIL_EXISTS` if the latter user tries to revert their email. Updated the code here to do the same.

Looks like the behaviour is the same even if `allowDuplicateEmails` is set to true.